cloudinit/net/cmdline.py

```diff
@@ -233,34 +235,31 @@ def read_initramfs_config():
     return None
 
 
-def _decomp_gzip(blob, strict=True):
-    # decompress blob. raise exception if not compressed unless strict=False.
+def _decomp_gzip(blob):
+    # decompress blob or return original blob
     with io.BytesIO(blob) as iobuf:
         gzfp = None
         try:
             gzfp = gzip.GzipFile(mode="rb", fileobj=iobuf)
             return gzfp.read()
         except IOError:
-            if strict:
-                raise
             return blob
         finally:
             if gzfp:
                 gzfp.close()
 
 
-def _b64dgz(b64str, gzipped="try"):
-    # decode a base64 string.  If gzipped is true, transparently uncompresss
-    # if gzipped is 'try', then try gunzip, returning the original on fail.
-    try:
-        blob = base64.b64decode(b64str)
-    except TypeError:
-        raise ValueError("Invalid base64 text: %s" % b64str)
+def _b64dgz(data):
+    """Decode a string base64 encoding, if gzipped, uncompress as well
 
-    if not gzipped:
-        return blob
+    :return: decompressed unencoded string of the data
+    """
+    try:
+        blob = base64.b64decode(data)
+    except (TypeError, ValueError):
+        raise ValueError("Invalid base64 text: %s" % data)
```

Comment:
I wonder if we we should log the error and continue?  Thoughts?

Reply:
I think that makes sense to me. At least try to bring the machine network up if we can and log an error in the event that it doesn't come up the way folks expected.

Reply:
OK, that was what I was feeling as well.

Reply:
so, if we try to pass here instead of raise that'd give us the ability to also support gzipped but not b64encoded content I think